kodilla-patterns/src/main/java/com/kodilla/patterns/prototype/library/Library.java

```diff
@@ -0,0 +1,48 @@
+package com.kodilla.patterns.prototype.library;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public final class Library extends Prototype<Library>{
+    private String name;
+    private Set<Book> books = new HashSet<>();
+
+    public Library(String name) {
+        this.name = name;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public Set<Book> getBooks() {
+        return books;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public void setBooks(Set<Book> books) {
+        this.books = books;
+    }
+
+    public Library shallowCopy() throws CloneNotSupportedException {
+        return super.clone();
+    }
+
+    public Library deepCopy() throws CloneNotSupportedException {
+        Library clonedLibrary = super.clone();
+        clonedLibrary.books = new HashSet<>();
+        for (Book book : books) {
+            clonedLibrary.getBooks().add(book);
```

Comment:
i tu nie bedzie deep copy. Jesli zmienie ksiazke w jednej bibliotece, to bedzie zmieniona rowniez w tej drugiej. Pamietaj, ze w Javie mamy przekazywanie argumentow do metod przez wartosc referencji.
np. Book book = new Book("Ogniem i mieczem", "Sienkiewicz", 1913);
      List<Book> listA = Arrays.asList(book);
      List<Book> listB = Arrays.asList(book);
      listA.get(0).author = "Mickiewicz";
      listB.get(0).getAuthor() ====> "Mickiewicz"